diff --git a/packaging/pep517_backend/_backend.py b/packaging/pep517_backend/_backend.py
index 72fa134..89d16f5 100644
--- a/packaging/pep517_backend/_backend.py
+++ b/packaging/pep517_backend/_backend.py
@@ -374,7 +374,7 @@ def get_requires_for_build_wheel(
         )
 
     c_ext_build_deps = [] if is_pure_python_build else [
-        'Cython ~= 3.0.0; python_version >= "3.12"',
+        'Cython >= 3.0.0, < 4.0.0; python_version >= "3.12"',
         'Cython; python_version < "3.12"',
     ]
 
@@ -388,4 +388,4 @@ get_requires_for_build_editable = get_requires_for_build_wheel
 prepare_metadata_for_build_wheel = patched_dist_get_long_description()(
     _setuptools_prepare_metadata_for_build_wheel,
 )
-prepare_metadata_for_build_editable = prepare_metadata_for_build_wheel
+prepare_metadata_for_build_editable = prepare_metadata_for_build_wheel
\ No newline at end of file
diff --git a/yarl.egg-info/PKG-INFO b/yarl.egg-info/PKG-INFO
index a4a7bf6..b9fd7c8 100644
--- a/yarl.egg-info/PKG-INFO
+++ b/yarl.egg-info/PKG-INFO
@@ -1,4 +1,4 @@
-Metadata-Version: 2.1
+Metadata-Version: 2.4
 Name: yarl
 Version: 1.18.3
 Summary: Yet another URL library
@@ -38,6 +38,7 @@ License-File: NOTICE
 Requires-Dist: idna>=2.0
 Requires-Dist: multidict>=4.0
 Requires-Dist: propcache>=0.2.0
+Dynamic: license-file
 
 yarl
 ====
@@ -274,19 +275,19 @@ Changelog
 Bug fixes
 ---------
 
-- Fixed uppercase ASCII hosts being rejected by ``URL.build()()`` and ``yarl.URL.with_host()`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Fixed uppercase ASCII hosts being rejected by :meth:`URL.build() <yarl.URL.build>` and :py:meth:`~yarl.URL.with_host` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#954 <https://github.com/aio-libs/yarl/issues/954>`__, `#1442 <https://github.com/aio-libs/yarl/issues/1442>`__.
+  :issue:`954`, :issue:`1442`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performances of multiple path properties on cache miss -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performances of multiple path properties on cache miss -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1443 <https://github.com/aio-libs/yarl/issues/1443>`__.
+  :issue:`1443`.
 
 
 ----
@@ -313,15 +314,15 @@ No significant changes.
 Miscellaneous internal changes
 ------------------------------
 
-- Improved cache performance when ``~yarl.URL`` objects are constructed from ``yarl.URL.build()`` with ``encoded=True`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved cache performance when :class:`~yarl.URL` objects are constructed from :py:meth:`~yarl.URL.build` with ``encoded=True`` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1432 <https://github.com/aio-libs/yarl/issues/1432>`__.
+  :issue:`1432`.
 
-- Improved cache performance for operations that produce a new ``~yarl.URL`` object -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved cache performance for operations that produce a new :class:`~yarl.URL` object -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1434 <https://github.com/aio-libs/yarl/issues/1434>`__, `#1436 <https://github.com/aio-libs/yarl/issues/1436>`__.
+  :issue:`1434`, :issue:`1436`.
 
 
 ----
@@ -336,28 +337,28 @@ Miscellaneous internal changes
 Features
 --------
 
-- Added ``keep_query`` and ``keep_fragment`` flags in the ``yarl.URL.with_path()``, ``yarl.URL.with_name()`` and ``yarl.URL.with_suffix()`` methods, allowing users to optionally retain the query string and fragment in the resulting URL when replacing the path -- by `@paul-nameless <https://github.com/sponsors/paul-nameless>`__.
+- Added ``keep_query`` and ``keep_fragment`` flags in the :py:meth:`yarl.URL.with_path`, :py:meth:`yarl.URL.with_name` and :py:meth:`yarl.URL.with_suffix` methods, allowing users to optionally retain the query string and fragment in the resulting URL when replacing the path -- by :user:`paul-nameless`.
 
   *Related issues and pull requests on GitHub:*
-  `#111 <https://github.com/aio-libs/yarl/issues/111>`__, `#1421 <https://github.com/aio-libs/yarl/issues/1421>`__.
+  :issue:`111`, :issue:`1421`.
 
 
 Contributor-facing changes
 --------------------------
 
-- Started running downstream ``aiohttp`` tests in CI -- by `@Cycloctane <https://github.com/sponsors/Cycloctane>`__.
+- Started running downstream ``aiohttp`` tests in CI -- by :user:`Cycloctane`.
 
   *Related issues and pull requests on GitHub:*
-  `#1415 <https://github.com/aio-libs/yarl/issues/1415>`__.
+  :issue:`1415`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of converting ``~yarl.URL`` to a string -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of converting :class:`~yarl.URL` to a string -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1422 <https://github.com/aio-libs/yarl/issues/1422>`__.
+  :issue:`1422`.
 
 
 ----
@@ -373,47 +374,47 @@ Bug fixes
 ---------
 
 - Stopped implicitly allowing the use of Cython pre-release versions when
-  building the distribution package -- by `@ajsanchezsanz <https://github.com/sponsors/ajsanchezsanz>`__ and
-  `@markgreene74 <https://github.com/sponsors/markgreene74>`__.
+  building the distribution package -- by :user:`ajsanchezsanz` and
+  :user:`markgreene74`.
 
   *Related issues and pull requests on GitHub:*
-  `#1411 <https://github.com/aio-libs/yarl/issues/1411>`__, `#1412 <https://github.com/aio-libs/yarl/issues/1412>`__.
+  :issue:`1411`, :issue:`1412`.
 
-- Fixed a bug causing ``~yarl.URL.port`` to return the default port when the given port was zero
-  -- by `@gmacon <https://github.com/sponsors/gmacon>`__.
+- Fixed a bug causing :attr:`~yarl.URL.port` to return the default port when the given port was zero
+  -- by :user:`gmacon`.
 
   *Related issues and pull requests on GitHub:*
-  `#1413 <https://github.com/aio-libs/yarl/issues/1413>`__.
+  :issue:`1413`.
 
 
 Features
 --------
 
-- Make error messages include details of incorrect type when ``port`` is not int in ``yarl.URL.build()``.
-  -- by `@Cycloctane <https://github.com/sponsors/Cycloctane>`__.
+- Make error messages include details of incorrect type when ``port`` is not int in :py:meth:`~yarl.URL.build`.
+  -- by :user:`Cycloctane`.
 
   *Related issues and pull requests on GitHub:*
-  `#1414 <https://github.com/aio-libs/yarl/issues/1414>`__.
+  :issue:`1414`.
 
 
 Packaging updates and notes for downstreams
 -------------------------------------------
 
 - Stopped implicitly allowing the use of Cython pre-release versions when
-  building the distribution package -- by `@ajsanchezsanz <https://github.com/sponsors/ajsanchezsanz>`__ and
-  `@markgreene74 <https://github.com/sponsors/markgreene74>`__.
+  building the distribution package -- by :user:`ajsanchezsanz` and
+  :user:`markgreene74`.
 
   *Related issues and pull requests on GitHub:*
-  `#1411 <https://github.com/aio-libs/yarl/issues/1411>`__, `#1412 <https://github.com/aio-libs/yarl/issues/1412>`__.
+  :issue:`1411`, :issue:`1412`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of the ``yarl.URL.joinpath()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.joinpath` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1418 <https://github.com/aio-libs/yarl/issues/1418>`__.
+  :issue:`1418`.
 
 
 ----
@@ -428,15 +429,15 @@ Miscellaneous internal changes
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of many ``~yarl.URL`` methods -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of many :class:`~yarl.URL` methods -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1396 <https://github.com/aio-libs/yarl/issues/1396>`__, `#1397 <https://github.com/aio-libs/yarl/issues/1397>`__, `#1398 <https://github.com/aio-libs/yarl/issues/1398>`__.
+  :issue:`1396`, :issue:`1397`, :issue:`1398`.
 
-- Improved performance of passing a `dict` or `str` to ``yarl.URL.extend_query()`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of passing a `dict` or `str` to :py:meth:`~yarl.URL.extend_query` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1401 <https://github.com/aio-libs/yarl/issues/1401>`__.
+  :issue:`1401`.
 
 
 ----
@@ -451,10 +452,10 @@ Miscellaneous internal changes
 Features
 --------
 
-- Added ``~yarl.URL.host_port_subcomponent`` which returns the ``3986#section-3.2.2`` host and ``3986#section-3.2.3`` port subcomponent -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Added :attr:`~yarl.URL.host_port_subcomponent` which returns the :rfc:`3986#section-3.2.2` host and :rfc:`3986#section-3.2.3` port subcomponent -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1375 <https://github.com/aio-libs/yarl/issues/1375>`__.
+  :issue:`1375`.
 
 
 ----
@@ -469,40 +470,40 @@ Features
 Bug fixes
 ---------
 
-- Fixed blocking I/O to load Python code when creating a new ``~yarl.URL`` with non-ascii characters in the network location part -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Fixed blocking I/O to load Python code when creating a new :class:`~yarl.URL` with non-ascii characters in the network location part -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1342 <https://github.com/aio-libs/yarl/issues/1342>`__.
+  :issue:`1342`.
 
 
 Removals and backward incompatible breaking changes
 ---------------------------------------------------
 
-- Migrated to using a single cache for encoding hosts -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Migrated to using a single cache for encoding hosts -- by :user:`bdraco`.
 
-  Passing ``ip_address_size`` and ``host_validate_size`` to ``yarl.cache_configure()`` is deprecated in favor of the new ``encode_host_size`` parameter and will be removed in a future release. For backwards compatibility, the old parameters affect the ``encode_host`` cache size.
+  Passing ``ip_address_size`` and ``host_validate_size`` to :py:meth:`~yarl.cache_configure` is deprecated in favor of the new ``encode_host_size`` parameter and will be removed in a future release. For backwards compatibility, the old parameters affect the ``encode_host`` cache size.
 
   *Related issues and pull requests on GitHub:*
-  `#1348 <https://github.com/aio-libs/yarl/issues/1348>`__, `#1357 <https://github.com/aio-libs/yarl/issues/1357>`__, `#1363 <https://github.com/aio-libs/yarl/issues/1363>`__.
+  :issue:`1348`, :issue:`1357`, :issue:`1363`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of constructing ``~yarl.URL`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of constructing :class:`~yarl.URL` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1336 <https://github.com/aio-libs/yarl/issues/1336>`__.
+  :issue:`1336`.
 
-- Improved performance of calling ``yarl.URL.build()`` and constructing unencoded ``~yarl.URL`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of calling :py:meth:`~yarl.URL.build` and constructing unencoded :class:`~yarl.URL` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1345 <https://github.com/aio-libs/yarl/issues/1345>`__.
+  :issue:`1345`.
 
-- Reworked the internal encoding cache to improve performance on cache hit -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Reworked the internal encoding cache to improve performance on cache hit -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1369 <https://github.com/aio-libs/yarl/issues/1369>`__.
+  :issue:`1369`.
 
 
 ----
@@ -517,80 +518,80 @@ Miscellaneous internal changes
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of the ``yarl.URL.joinpath()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.joinpath` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1304 <https://github.com/aio-libs/yarl/issues/1304>`__.
+  :issue:`1304`.
 
-- Improved performance of the ``yarl.URL.extend_query()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.extend_query` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1305 <https://github.com/aio-libs/yarl/issues/1305>`__.
+  :issue:`1305`.
 
-- Improved performance of the ``yarl.URL.origin()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.origin` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1306 <https://github.com/aio-libs/yarl/issues/1306>`__.
+  :issue:`1306`.
 
-- Improved performance of the ``yarl.URL.with_path()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.with_path` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1307 <https://github.com/aio-libs/yarl/issues/1307>`__.
+  :issue:`1307`.
 
-- Improved performance of the ``yarl.URL.with_query()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.with_query` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1308 <https://github.com/aio-libs/yarl/issues/1308>`__, `#1328 <https://github.com/aio-libs/yarl/issues/1328>`__.
+  :issue:`1308`, :issue:`1328`.
 
-- Improved performance of the ``yarl.URL.update_query()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.update_query` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1309 <https://github.com/aio-libs/yarl/issues/1309>`__, `#1327 <https://github.com/aio-libs/yarl/issues/1327>`__.
+  :issue:`1309`, :issue:`1327`.
 
-- Improved performance of the ``yarl.URL.join()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.join` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1313 <https://github.com/aio-libs/yarl/issues/1313>`__.
+  :issue:`1313`.
 
-- Improved performance of ``~yarl.URL`` equality checks -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of :class:`~yarl.URL` equality checks -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1315 <https://github.com/aio-libs/yarl/issues/1315>`__.
+  :issue:`1315`.
 
-- Improved performance of ``~yarl.URL`` methods that modify the network location -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of :class:`~yarl.URL` methods that modify the network location -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1316 <https://github.com/aio-libs/yarl/issues/1316>`__.
+  :issue:`1316`.
 
-- Improved performance of the ``yarl.URL.with_fragment()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.with_fragment` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1317 <https://github.com/aio-libs/yarl/issues/1317>`__.
+  :issue:`1317`.
 
-- Improved performance of calculating the hash of ``~yarl.URL`` objects -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of calculating the hash of :class:`~yarl.URL` objects -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1318 <https://github.com/aio-libs/yarl/issues/1318>`__.
+  :issue:`1318`.
 
-- Improved performance of the ``yarl.URL.relative()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.relative` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1319 <https://github.com/aio-libs/yarl/issues/1319>`__.
+  :issue:`1319`.
 
-- Improved performance of the ``yarl.URL.with_name()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.with_name` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1320 <https://github.com/aio-libs/yarl/issues/1320>`__.
+  :issue:`1320`.
 
-- Improved performance of ``~yarl.URL.parent`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of :attr:`~yarl.URL.parent` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1321 <https://github.com/aio-libs/yarl/issues/1321>`__.
+  :issue:`1321`.
 
-- Improved performance of the ``yarl.URL.with_scheme()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.with_scheme` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1322 <https://github.com/aio-libs/yarl/issues/1322>`__.
+  :issue:`1322`.
 
 
 ----
@@ -605,20 +606,20 @@ Miscellaneous internal changes
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of the quoter when all characters are safe -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the quoter when all characters are safe -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1288 <https://github.com/aio-libs/yarl/issues/1288>`__.
+  :issue:`1288`.
 
-- Improved performance of unquoting strings -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of unquoting strings -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1292 <https://github.com/aio-libs/yarl/issues/1292>`__, `#1293 <https://github.com/aio-libs/yarl/issues/1293>`__.
+  :issue:`1292`, :issue:`1293`.
 
-- Improved performance of calling ``yarl.URL.build()`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of calling :py:meth:`~yarl.URL.build` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1297 <https://github.com/aio-libs/yarl/issues/1297>`__.
+  :issue:`1297`.
 
 
 ----
@@ -633,30 +634,30 @@ Miscellaneous internal changes
 Bug fixes
 ---------
 
-- Fixed ``yarl.URL.build()`` failing to validate paths must start with a ``/`` when passing ``authority`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Fixed :py:meth:`~yarl.URL.build` failing to validate paths must start with a ``/`` when passing ``authority`` -- by :user:`bdraco`.
 
   The validation only worked correctly when passing ``host``.
 
   *Related issues and pull requests on GitHub:*
-  `#1265 <https://github.com/aio-libs/yarl/issues/1265>`__.
+  :issue:`1265`.
 
 
 Removals and backward incompatible breaking changes
 ---------------------------------------------------
 
-- Removed support for Python 3.8 as it has reached end of life -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Removed support for Python 3.8 as it has reached end of life -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1203 <https://github.com/aio-libs/yarl/issues/1203>`__.
+  :issue:`1203`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of constructing ``~yarl.URL`` when the net location is only the host -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of constructing :class:`~yarl.URL` when the net location is only the host -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1271 <https://github.com/aio-libs/yarl/issues/1271>`__.
+  :issue:`1271`.
 
 
 ----
@@ -671,25 +672,25 @@ Miscellaneous internal changes
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of converting ``~yarl.URL`` to a string -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of converting :class:`~yarl.URL` to a string -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1234 <https://github.com/aio-libs/yarl/issues/1234>`__.
+  :issue:`1234`.
 
-- Improved performance of ``yarl.URL.joinpath()`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of :py:meth:`~yarl.URL.joinpath` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1248 <https://github.com/aio-libs/yarl/issues/1248>`__, `#1250 <https://github.com/aio-libs/yarl/issues/1250>`__.
+  :issue:`1248`, :issue:`1250`.
 
-- Improved performance of constructing query strings from ``~multidict.MultiDict`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of constructing query strings from :class:`~multidict.MultiDict` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1256 <https://github.com/aio-libs/yarl/issues/1256>`__.
+  :issue:`1256`.
 
-- Improved performance of constructing query strings with ``int`` values -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of constructing query strings with ``int`` values -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1259 <https://github.com/aio-libs/yarl/issues/1259>`__.
+  :issue:`1259`.
 
 
 ----
@@ -704,20 +705,20 @@ Miscellaneous internal changes
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of calling ``yarl.URL.build()`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of calling :py:meth:`~yarl.URL.build` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1222 <https://github.com/aio-libs/yarl/issues/1222>`__.
+  :issue:`1222`.
 
-- Improved performance of all ``~yarl.URL`` methods that create new ``~yarl.URL`` objects -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of all :class:`~yarl.URL` methods that create new :class:`~yarl.URL` objects -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1226 <https://github.com/aio-libs/yarl/issues/1226>`__.
+  :issue:`1226`.
 
-- Improved performance of ``~yarl.URL`` methods that modify the network location -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of :class:`~yarl.URL` methods that modify the network location -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1229 <https://github.com/aio-libs/yarl/issues/1229>`__.
+  :issue:`1229`.
 
 
 ----
@@ -732,43 +733,43 @@ Miscellaneous internal changes
 Bug fixes
 ---------
 
-- Fixed validation with ``yarl.URL.with_scheme()`` when passed scheme is not lowercase -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Fixed validation with :py:meth:`~yarl.URL.with_scheme` when passed scheme is not lowercase -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1189 <https://github.com/aio-libs/yarl/issues/1189>`__.
+  :issue:`1189`.
 
 
 Features
 --------
 
-- Started building ``armv7l`` wheels -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Started building ``armv7l`` wheels -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1204 <https://github.com/aio-libs/yarl/issues/1204>`__.
+  :issue:`1204`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of constructing unencoded ``~yarl.URL`` objects -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of constructing unencoded :class:`~yarl.URL` objects -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1188 <https://github.com/aio-libs/yarl/issues/1188>`__.
+  :issue:`1188`.
 
-- Added a cache for parsing hosts to reduce overhead of encoding ``~yarl.URL`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Added a cache for parsing hosts to reduce overhead of encoding :class:`~yarl.URL` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1190 <https://github.com/aio-libs/yarl/issues/1190>`__.
+  :issue:`1190`.
 
-- Improved performance of constructing query strings from ``~collections.abc.Mapping`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of constructing query strings from :class:`~collections.abc.Mapping` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1193 <https://github.com/aio-libs/yarl/issues/1193>`__.
+  :issue:`1193`.
 
-- Improved performance of converting ``~yarl.URL`` objects to strings -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of converting :class:`~yarl.URL` objects to strings -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1198 <https://github.com/aio-libs/yarl/issues/1198>`__.
+  :issue:`1198`.
 
 
 ----
@@ -783,50 +784,50 @@ Miscellaneous internal changes
 Packaging updates and notes for downstreams
 -------------------------------------------
 
-- Switched to using the ``propcache`` package for property caching
-  -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Switched to using the :mod:`propcache <propcache.api>` package for property caching
+  -- by :user:`bdraco`.
 
-  The ``propcache`` package is derived from the property caching
-  code in ``yarl`` and has been broken out to avoid maintaining it for multiple
+  The :mod:`propcache <propcache.api>` package is derived from the property caching
+  code in :mod:`yarl` and has been broken out to avoid maintaining it for multiple
   projects.
 
   *Related issues and pull requests on GitHub:*
-  `#1169 <https://github.com/aio-libs/yarl/issues/1169>`__.
+  :issue:`1169`.
 
 
 Contributor-facing changes
 --------------------------
 
-- Started testing with Hypothesis -- by `@webknjaz <https://github.com/sponsors/webknjaz>`__ and `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Started testing with Hypothesis -- by :user:`webknjaz` and :user:`bdraco`.
 
-  Special thanks to `@Zac-HD <https://github.com/sponsors/Zac-HD>`__ for helping us get started with this framework.
+  Special thanks to :user:`Zac-HD` for helping us get started with this framework.
 
   *Related issues and pull requests on GitHub:*
-  `#860 <https://github.com/aio-libs/yarl/issues/860>`__.
+  :issue:`860`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of ``yarl.URL.is_default_port()`` when no explicit port is set -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of :py:meth:`~yarl.URL.is_default_port` when no explicit port is set -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1168 <https://github.com/aio-libs/yarl/issues/1168>`__.
+  :issue:`1168`.
 
-- Improved performance of converting ``~yarl.URL`` to a string when no explicit port is set -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of converting :class:`~yarl.URL` to a string when no explicit port is set -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1170 <https://github.com/aio-libs/yarl/issues/1170>`__.
+  :issue:`1170`.
 
-- Improved performance of the ``yarl.URL.origin()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :py:meth:`~yarl.URL.origin` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1175 <https://github.com/aio-libs/yarl/issues/1175>`__.
+  :issue:`1175`.
 
-- Improved performance of encoding hosts -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of encoding hosts -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1176 <https://github.com/aio-libs/yarl/issues/1176>`__.
+  :issue:`1176`.
 
 
 ----
@@ -841,10 +842,10 @@ Miscellaneous internal changes
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of calling ``yarl.URL.build()`` with ``authority`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of calling :py:meth:`~yarl.URL.build` with ``authority`` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1163 <https://github.com/aio-libs/yarl/issues/1163>`__.
+  :issue:`1163`.
 
 
 ----
@@ -861,26 +862,26 @@ Bug fixes
 
 - Started rejecting ASCII hostnames with invalid characters. For host strings that
   look like authority strings, the exception message includes advice on what to do
-  instead -- by `@mjpieters <https://github.com/sponsors/mjpieters>`__.
+  instead -- by :user:`mjpieters`.
 
   *Related issues and pull requests on GitHub:*
-  `#880 <https://github.com/aio-libs/yarl/issues/880>`__, `#954 <https://github.com/aio-libs/yarl/issues/954>`__.
+  :issue:`880`, :issue:`954`.
 
-- Fixed IPv6 addresses missing brackets when the ``~yarl.URL`` was converted to a string -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Fixed IPv6 addresses missing brackets when the :class:`~yarl.URL` was converted to a string -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1157 <https://github.com/aio-libs/yarl/issues/1157>`__, `#1158 <https://github.com/aio-libs/yarl/issues/1158>`__.
+  :issue:`1157`, :issue:`1158`.
 
 
 Features
 --------
 
-- Added ``~yarl.URL.host_subcomponent`` which returns the ``3986#section-3.2.2`` host subcomponent -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Added :attr:`~yarl.URL.host_subcomponent` which returns the :rfc:`3986#section-3.2.2` host subcomponent -- by :user:`bdraco`.
 
-  The only current practical difference between ``~yarl.URL.raw_host`` and ``~yarl.URL.host_subcomponent`` is that IPv6 addresses are returned bracketed.
+  The only current practical difference between :attr:`~yarl.URL.raw_host` and :attr:`~yarl.URL.host_subcomponent` is that IPv6 addresses are returned bracketed.
 
   *Related issues and pull requests on GitHub:*
-  `#1159 <https://github.com/aio-libs/yarl/issues/1159>`__.
+  :issue:`1159`.
 
 
 ----
@@ -907,30 +908,30 @@ No significant changes.
 Features
 --------
 
-- Added ``~yarl.URL.path_safe`` to be able to fetch the path without ``%2F`` and ``%25`` decoded -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Added :attr:`~yarl.URL.path_safe` to be able to fetch the path without ``%2F`` and ``%25`` decoded -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1150 <https://github.com/aio-libs/yarl/issues/1150>`__.
+  :issue:`1150`.
 
 
 Removals and backward incompatible breaking changes
 ---------------------------------------------------
 
-- Restore decoding ``%2F`` (``/``) in ``URL.path`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Restore decoding ``%2F`` (``/``) in ``URL.path`` -- by :user:`bdraco`.
 
-  This change restored the behavior before `#1057 <https://github.com/aio-libs/yarl/issues/1057>`__.
+  This change restored the behavior before :issue:`1057`.
 
   *Related issues and pull requests on GitHub:*
-  `#1151 <https://github.com/aio-libs/yarl/issues/1151>`__.
+  :issue:`1151`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of processing paths -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of processing paths -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1143 <https://github.com/aio-libs/yarl/issues/1143>`__.
+  :issue:`1143`.
 
 
 ----
@@ -945,33 +946,33 @@ Miscellaneous internal changes
 Bug fixes
 ---------
 
-- Allowed scheme replacement for relative URLs if the scheme does not require a host -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Allowed scheme replacement for relative URLs if the scheme does not require a host -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#280 <https://github.com/aio-libs/yarl/issues/280>`__, `#1138 <https://github.com/aio-libs/yarl/issues/1138>`__.
+  :issue:`280`, :issue:`1138`.
 
-- Allowed empty host for URL schemes other than the special schemes listed in the WHATWG URL spec -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Allowed empty host for URL schemes other than the special schemes listed in the WHATWG URL spec -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1136 <https://github.com/aio-libs/yarl/issues/1136>`__.
+  :issue:`1136`.
 
 
 Features
 --------
 
-- Loosened restriction on integers as query string values to allow classes that implement ``__int__`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Loosened restriction on integers as query string values to allow classes that implement ``__int__`` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1139 <https://github.com/aio-libs/yarl/issues/1139>`__.
+  :issue:`1139`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of normalizing paths -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of normalizing paths -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1137 <https://github.com/aio-libs/yarl/issues/1137>`__.
+  :issue:`1137`.
 
 
 ----
@@ -986,46 +987,46 @@ Miscellaneous internal changes
 Features
 --------
 
-- Added ``URL.extend_query()()`` method, which can be used to extend parameters without replacing same named keys -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Added :meth:`URL.extend_query() <yarl.URL.extend_query>` method, which can be used to extend parameters without replacing same named keys -- by :user:`bdraco`.
 
   This method was primarily added to replace the inefficient hand rolled method currently used in ``aiohttp``.
 
   *Related issues and pull requests on GitHub:*
-  `#1128 <https://github.com/aio-libs/yarl/issues/1128>`__.
+  :issue:`1128`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of the Cython ``cached_property`` implementation -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the Cython ``cached_property`` implementation -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1122 <https://github.com/aio-libs/yarl/issues/1122>`__.
+  :issue:`1122`.
 
-- Simplified computing ports by removing unnecessary code -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Simplified computing ports by removing unnecessary code -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1123 <https://github.com/aio-libs/yarl/issues/1123>`__.
+  :issue:`1123`.
 
-- Improved performance of encoding non IPv6 hosts -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of encoding non IPv6 hosts -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1125 <https://github.com/aio-libs/yarl/issues/1125>`__.
+  :issue:`1125`.
 
-- Improved performance of ``URL.build()()`` when the path, query string, or fragment is an empty string -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of :meth:`URL.build() <yarl.URL.build>` when the path, query string, or fragment is an empty string -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1126 <https://github.com/aio-libs/yarl/issues/1126>`__.
+  :issue:`1126`.
 
-- Improved performance of the ``URL.update_query()()`` method -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of the :meth:`URL.update_query() <yarl.URL.update_query>` method -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1130 <https://github.com/aio-libs/yarl/issues/1130>`__.
+  :issue:`1130`.
 
-- Improved performance of processing query string changes when arguments are ``str`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of processing query string changes when arguments are :class:`str` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1131 <https://github.com/aio-libs/yarl/issues/1131>`__.
+  :issue:`1131`.
 
 
 ----
@@ -1040,49 +1041,49 @@ Miscellaneous internal changes
 Bug fixes
 ---------
 
-- Fixed joining a path when the existing path was empty -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Fixed joining a path when the existing path was empty -- by :user:`bdraco`.
 
-  A regression in ``URL.join()()`` was introduced in `#1082 <https://github.com/aio-libs/yarl/issues/1082>`__.
+  A regression in :meth:`URL.join() <yarl.URL.join>` was introduced in :issue:`1082`.
 
   *Related issues and pull requests on GitHub:*
-  `#1118 <https://github.com/aio-libs/yarl/issues/1118>`__.
+  :issue:`1118`.
 
 
 Features
 --------
 
-- Added ``URL.without_query_params()()`` method, to drop some parameters from query string -- by `@hongquan <https://github.com/sponsors/hongquan>`__.
+- Added :meth:`URL.without_query_params() <yarl.URL.without_query_params>` method, to drop some parameters from query string -- by :user:`hongquan`.
 
   *Related issues and pull requests on GitHub:*
-  `#774 <https://github.com/aio-libs/yarl/issues/774>`__, `#898 <https://github.com/aio-libs/yarl/issues/898>`__, `#1010 <https://github.com/aio-libs/yarl/issues/1010>`__.
+  :issue:`774`, :issue:`898`, :issue:`1010`.
 
-- The previously protected types ``_SimpleQuery``, ``_QueryVariable``, and ``_Query`` are now available for use externally as ``SimpleQuery``, ``QueryVariable``, and ``Query`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- The previously protected types ``_SimpleQuery``, ``_QueryVariable``, and ``_Query`` are now available for use externally as ``SimpleQuery``, ``QueryVariable``, and ``Query`` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1050 <https://github.com/aio-libs/yarl/issues/1050>`__, `#1113 <https://github.com/aio-libs/yarl/issues/1113>`__.
+  :issue:`1050`, :issue:`1113`.
 
 
 Contributor-facing changes
 --------------------------
 
-- Replaced all ``~typing.Optional`` with ``~typing.Union`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Replaced all :class:`~typing.Optional` with :class:`~typing.Union` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1095 <https://github.com/aio-libs/yarl/issues/1095>`__.
+  :issue:`1095`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Significantly improved performance of parsing the network location -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Significantly improved performance of parsing the network location -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1112 <https://github.com/aio-libs/yarl/issues/1112>`__.
+  :issue:`1112`.
 
-- Added internal types to the cache to prevent future refactoring errors -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Added internal types to the cache to prevent future refactoring errors -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1117 <https://github.com/aio-libs/yarl/issues/1117>`__.
+  :issue:`1117`.
 
 
 ----
@@ -1097,21 +1098,21 @@ Miscellaneous internal changes
 Bug fixes
 ---------
 
-- Fixed a ``TypeError`` with ``MultiDictProxy`` and Python 3.8 -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Fixed a :exc:`TypeError` with ``MultiDictProxy`` and Python 3.8 -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1084 <https://github.com/aio-libs/yarl/issues/1084>`__, `#1105 <https://github.com/aio-libs/yarl/issues/1105>`__, `#1107 <https://github.com/aio-libs/yarl/issues/1107>`__.
+  :issue:`1084`, :issue:`1105`, :issue:`1107`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of encoding hosts -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of encoding hosts -- by :user:`bdraco`.
 
-  Previously, the library would unconditionally try to parse a host as an IP Address. The library now avoids trying to parse a host as an IP Address if the string is not in one of the formats described in ``3986#section-3.2.2``.
+  Previously, the library would unconditionally try to parse a host as an IP Address. The library now avoids trying to parse a host as an IP Address if the string is not in one of the formats described in :rfc:`3986#section-3.2.2`.
 
   *Related issues and pull requests on GitHub:*
-  `#1104 <https://github.com/aio-libs/yarl/issues/1104>`__.
+  :issue:`1104`.
 
 
 ----
@@ -1126,15 +1127,15 @@ Miscellaneous internal changes
 Bug fixes
 ---------
 
-- ``URL.join()()`` has been changed to match
-  ``3986`` and align with
-  ``/ operation()`` and ``URL.joinpath()()``
+- :meth:`URL.join() <yarl.URL.join>` has been changed to match
+  :rfc:`3986` and align with
+  :meth:`/ operation <yarl.URL.__truediv__>` and :meth:`URL.joinpath() <yarl.URL.joinpath>`
   when joining URLs with empty segments.
-  Previously ``urllib.parse.urljoin`` was used,
+  Previously :py:func:`urllib.parse.urljoin` was used,
   which has known issues with empty segments
   (`python/cpython#84774 <https://github.com/python/cpython/issues/84774>`_).
 
-  Due to the semantics of ``URL.join()()``, joining an
+  Due to the semantics of :meth:`URL.join() <yarl.URL.join>`, joining an
   URL with scheme requires making it relative, prefixing with ``./``.
 
   .. code-block:: pycon
@@ -1152,21 +1153,21 @@ Bug fixes
 
 
 
-  -- by `@commonism <https://github.com/sponsors/commonism>`__
+  -- by :user:`commonism`
 
   This change initially appeared in 1.9.5 but was reverted in 1.9.6 to resolve a problem with query string handling.
 
   *Related issues and pull requests on GitHub:*
-  `#1039 <https://github.com/aio-libs/yarl/issues/1039>`__, `#1082 <https://github.com/aio-libs/yarl/issues/1082>`__.
+  :issue:`1039`, :issue:`1082`.
 
 
 Features
 --------
 
-- Added ``~yarl.URL.absolute`` which is now preferred over ``URL.is_absolute()`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Added :attr:`~yarl.URL.absolute` which is now preferred over ``URL.is_absolute()`` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1100 <https://github.com/aio-libs/yarl/issues/1100>`__.
+  :issue:`1100`.
 
 
 ----
@@ -1181,10 +1182,10 @@ Features
 Bug fixes
 ---------
 
-- Added missing type on ``~yarl.URL.port`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Added missing type on :attr:`~yarl.URL.port` -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1097 <https://github.com/aio-libs/yarl/issues/1097>`__.
+  :issue:`1097`.
 
 
 ----
@@ -1199,33 +1200,33 @@ Bug fixes
 Features
 --------
 
-- Covered the ``~yarl.URL`` object with types -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Covered the :class:`~yarl.URL` object with types -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1084 <https://github.com/aio-libs/yarl/issues/1084>`__.
+  :issue:`1084`.
 
-- Cache parsing of IP Addresses when encoding hosts -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Cache parsing of IP Addresses when encoding hosts -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1086 <https://github.com/aio-libs/yarl/issues/1086>`__.
+  :issue:`1086`.
 
 
 Contributor-facing changes
 --------------------------
 
-- Covered the ``~yarl.URL`` object with types -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Covered the :class:`~yarl.URL` object with types -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1084 <https://github.com/aio-libs/yarl/issues/1084>`__.
+  :issue:`1084`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of handling ports -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of handling ports -- by :user:`bdraco`.
 
   *Related issues and pull requests on GitHub:*
-  `#1081 <https://github.com/aio-libs/yarl/issues/1081>`__.
+  :issue:`1081`.
 
 
 ----
@@ -1240,23 +1241,23 @@ Miscellaneous internal changes
 Removals and backward incompatible breaking changes
 ---------------------------------------------------
 
-- Removed support ``3986#section-3.2.3`` port normalization when the scheme is not one of ``http``, ``https``, ``wss``, or ``ws`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Removed support :rfc:`3986#section-3.2.3` port normalization when the scheme is not one of ``http``, ``https``, ``wss``, or ``ws`` -- by :user:`bdraco`.
 
-  Support for port normalization was recently added in `#1033 <https://github.com/aio-libs/yarl/issues/1033>`__ and contained code that would do blocking I/O if the scheme was not one of the four listed above. The code has been removed because this library is intended to be safe for usage with ``asyncio``.
+  Support for port normalization was recently added in :issue:`1033` and contained code that would do blocking I/O if the scheme was not one of the four listed above. The code has been removed because this library is intended to be safe for usage with :mod:`asyncio`.
 
   *Related issues and pull requests on GitHub:*
-  `#1076 <https://github.com/aio-libs/yarl/issues/1076>`__.
+  :issue:`1076`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- Improved performance of property caching -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- Improved performance of property caching -- by :user:`bdraco`.
 
   The ``reify`` implementation from ``aiohttp`` was adapted to replace the internal ``cached_property`` implementation.
 
   *Related issues and pull requests on GitHub:*
-  `#1070 <https://github.com/aio-libs/yarl/issues/1070>`__.
+  :issue:`1070`.
 
 
 ----
@@ -1271,12 +1272,12 @@ Miscellaneous internal changes
 Bug fixes
 ---------
 
-- Reverted ``3986`` compatible ``URL.join()()`` honoring empty segments which was introduced in `#1039 <https://github.com/aio-libs/yarl/issues/1039>`__.
+- Reverted :rfc:`3986` compatible :meth:`URL.join() <yarl.URL.join>` honoring empty segments which was introduced in :issue:`1039`.
 
   This change introduced a regression handling query string parameters with joined URLs. The change was reverted to maintain compatibility with the previous behavior.
 
   *Related issues and pull requests on GitHub:*
-  `#1067 <https://github.com/aio-libs/yarl/issues/1067>`__.
+  :issue:`1067`.
 
 
 ----
@@ -1292,7 +1293,7 @@ Bug fixes
 ---------
 
 - Joining URLs with empty segments has been changed
-  to match ``3986``.
+  to match :rfc:`3986`.
 
   Previously empty segments would be removed from path,
   breaking use-cases such as
@@ -1301,7 +1302,7 @@ Bug fixes
 
      URL("https://web.archive.org/web/") / "https://github.com/"
 
-  Now ``/ operation()`` and ``URL.joinpath()()``
+  Now :meth:`/ operation <yarl.URL.__truediv__>` and :meth:`URL.joinpath() <yarl.URL.joinpath>`
   keep empty segments, but do not introduce new empty segments.
   e.g.
 
@@ -1311,30 +1312,30 @@ Bug fixes
 
   does not introduce an empty segment.
 
-  -- by `@commonism <https://github.com/sponsors/commonism>`__ and `@youtux <https://github.com/sponsors/youtux>`__
+  -- by :user:`commonism` and :user:`youtux`
 
   *Related issues and pull requests on GitHub:*
-  `#1026 <https://github.com/aio-libs/yarl/issues/1026>`__.
+  :issue:`1026`.
 
 - The default protocol ports of well-known URI schemes are now taken into account
   during the normalization of the URL string representation in accordance with
-  ``3986#section-3.2.3``.
+  :rfc:`3986#section-3.2.3`.
 
-  Specified ports are removed from the ``str`` representation of a ``~yarl.URL``
-  if the port matches the scheme's default port -- by `@commonism <https://github.com/sponsors/commonism>`__.
+  Specified ports are removed from the :class:`str` representation of a :class:`~yarl.URL`
+  if the port matches the scheme's default port -- by :user:`commonism`.
 
   *Related issues and pull requests on GitHub:*
-  `#1033 <https://github.com/aio-libs/yarl/issues/1033>`__.
+  :issue:`1033`.
 
-- ``URL.join()()`` has been changed to match
-  ``3986`` and align with
-  ``/ operation()`` and ``URL.joinpath()()``
+- :meth:`URL.join() <yarl.URL.join>` has been changed to match
+  :rfc:`3986` and align with
+  :meth:`/ operation <yarl.URL.__truediv__>` and :meth:`URL.joinpath() <yarl.URL.joinpath>`
   when joining URLs with empty segments.
-  Previously ``urllib.parse.urljoin`` was used,
+  Previously :py:func:`urllib.parse.urljoin` was used,
   which has known issues with empty segments
   (`python/cpython#84774 <https://github.com/python/cpython/issues/84774>`_).
 
-  Due to the semantics of ``URL.join()()``, joining an
+  Due to the semantics of :meth:`URL.join() <yarl.URL.join>`, joining an
   URL with scheme requires making it relative, prefixing with ``./``.
 
   .. code-block:: pycon
@@ -1352,63 +1353,63 @@ Bug fixes
 
 
 
-  -- by `@commonism <https://github.com/sponsors/commonism>`__
+  -- by :user:`commonism`
 
   *Related issues and pull requests on GitHub:*
-  `#1039 <https://github.com/aio-libs/yarl/issues/1039>`__.
+  :issue:`1039`.
 
 
 Removals and backward incompatible breaking changes
 ---------------------------------------------------
 
 - Stopped decoding ``%2F`` (``/``) in ``URL.path``, as this could lead to code incorrectly treating it as a path separator
-  -- by `@Dreamsorcerer <https://github.com/sponsors/Dreamsorcerer>`__.
+  -- by :user:`Dreamsorcerer`.
 
   *Related issues and pull requests on GitHub:*
-  `#1057 <https://github.com/aio-libs/yarl/issues/1057>`__.
+  :issue:`1057`.
 
-- Dropped support for Python 3.7 -- by `@Dreamsorcerer <https://github.com/sponsors/Dreamsorcerer>`__.
+- Dropped support for Python 3.7 -- by :user:`Dreamsorcerer`.
 
   *Related issues and pull requests on GitHub:*
-  `#1016 <https://github.com/aio-libs/yarl/issues/1016>`__.
+  :issue:`1016`.
 
 
 Improved documentation
 ----------------------
 
-- On the ``Contributing docs`` page,
+- On the :doc:`Contributing docs <contributing/guidelines>` page,
   a link to the ``Towncrier philosophy`` has been fixed.
 
   *Related issues and pull requests on GitHub:*
-  `#981 <https://github.com/aio-libs/yarl/issues/981>`__.
+  :issue:`981`.
 
-- The pre-existing ``/ magic method()``
-  has been documented in the API reference -- by `@commonism <https://github.com/sponsors/commonism>`__.
+- The pre-existing :meth:`/ magic method <yarl.URL.__truediv__>`
+  has been documented in the API reference -- by :user:`commonism`.
 
   *Related issues and pull requests on GitHub:*
-  `#1026 <https://github.com/aio-libs/yarl/issues/1026>`__.
+  :issue:`1026`.
 
 
 Packaging updates and notes for downstreams
 -------------------------------------------
 
 - A flaw in the logic for copying the project directory into a
-  temporary folder that led to infinite recursion when ``TMPDIR``
+  temporary folder that led to infinite recursion when :envvar:`TMPDIR`
   was set to a project subdirectory path. This was happening in Fedora
   and its downstream due to the use of `pyproject-rpm-macros
   <https://src.fedoraproject.org/rpms/pyproject-rpm-macros>`__. It was
   only reproducible with ``pip wheel`` and was not affecting the
   ``pyproject-build`` users.
 
-  -- by `@hroncok <https://github.com/sponsors/hroncok>`__ and `@webknjaz <https://github.com/sponsors/webknjaz>`__
+  -- by :user:`hroncok` and :user:`webknjaz`
 
   *Related issues and pull requests on GitHub:*
-  `#992 <https://github.com/aio-libs/yarl/issues/992>`__, `#1014 <https://github.com/aio-libs/yarl/issues/1014>`__.
+  :issue:`992`, :issue:`1014`.
 
 - Support Python 3.13 and publish non-free-threaded wheels
 
   *Related issues and pull requests on GitHub:*
-  `#1054 <https://github.com/aio-libs/yarl/issues/1054>`__.
+  :issue:`1054`.
 
 
 Contributor-facing changes
@@ -1416,26 +1417,26 @@ Contributor-facing changes
 
 - The CI/CD setup has been updated to test ``arm64`` wheels
   under macOS 14, except for Python 3.7 that is unsupported
-  in that environment -- by `@webknjaz <https://github.com/sponsors/webknjaz>`__.
+  in that environment -- by :user:`webknjaz`.
 
   *Related issues and pull requests on GitHub:*
-  `#1015 <https://github.com/aio-libs/yarl/issues/1015>`__.
+  :issue:`1015`.
 
-- Removed unused type ignores and casts -- by `@hauntsaninja <https://github.com/sponsors/hauntsaninja>`__.
+- Removed unused type ignores and casts -- by :user:`hauntsaninja`.
 
   *Related issues and pull requests on GitHub:*
-  `#1031 <https://github.com/aio-libs/yarl/issues/1031>`__.
+  :issue:`1031`.
 
 
 Miscellaneous internal changes
 ------------------------------
 
-- ``port``, ``scheme``, and ``raw_host`` are now ``cached_property`` -- by `@bdraco <https://github.com/sponsors/bdraco>`__.
+- ``port``, ``scheme``, and ``raw_host`` are now ``cached_property`` -- by :user:`bdraco`.
 
-  ``aiohttp`` accesses these properties quite often, which cause ``urllib`` to build the ``_hostinfo`` property every time. ``port``, ``scheme``, and ``raw_host`` are now cached properties, which will improve performance.
+  ``aiohttp`` accesses these properties quite often, which cause :mod:`urllib` to build the ``_hostinfo`` property every time. ``port``, ``scheme``, and ``raw_host`` are now cached properties, which will improve performance.
 
   *Related issues and pull requests on GitHub:*
-  `#1044 <https://github.com/aio-libs/yarl/issues/1044>`__, `#1058 <https://github.com/aio-libs/yarl/issues/1058>`__.
+  :issue:`1044`, :issue:`1058`.
 
 
 ----
@@ -1447,18 +1448,18 @@ Miscellaneous internal changes
 Bug fixes
 ---------
 
-- Started raising ``TypeError`` when a string value is passed into
-  ``yarl.URL.build()`` as the ``port`` argument  -- by `@commonism <https://github.com/sponsors/commonism>`__.
+- Started raising :py:exc:`TypeError` when a string value is passed into
+  :py:meth:`~yarl.URL.build` as the ``port`` argument  -- by :user:`commonism`.
 
-  Previously the empty string as port would create malformed URLs when rendered as string representations. (`#883 <https://github.com/aio-libs/yarl/issues/883>`__)
+  Previously the empty string as port would create malformed URLs when rendered as string representations. (:issue:`883`)
 
 
 Packaging updates and notes for downstreams
 -------------------------------------------
 
-- The leading ``--`` has been dropped from the `PEP 517 <https://peps.python.org/pep-517>`__ in-tree build
+- The leading ``--`` has been dropped from the :pep:`517` in-tree build
   backend config setting names. ``--pure-python`` is now just ``pure-python``
-  -- by `@webknjaz <https://github.com/sponsors/webknjaz>`__.
+  -- by :user:`webknjaz`.
 
   The usage now looks as follows:
 
@@ -1468,18 +1469,18 @@ Packaging updates and notes for downstreams
           --config-setting=pure-python=true \
           --config-setting=with-cython-tracing=true
 
-  (`#963 <https://github.com/aio-libs/yarl/issues/963>`__)
+  (:issue:`963`)
 
 
 Contributor-facing changes
 --------------------------
 
-- A step-by-step ``Release Guide`` guide has
-  been added, describing how to release *yarl* -- by `@webknjaz <https://github.com/sponsors/webknjaz>`__.
+- A step-by-step :doc:`Release Guide <contributing/release_guide>` guide has
+  been added, describing how to release *yarl* -- by :user:`webknjaz`.
 
-  This is primarily targeting maintainers. (`#960 <https://github.com/aio-libs/yarl/issues/960>`__)
+  This is primarily targeting maintainers. (:issue:`960`)
 - Coverage collection has been implemented for the Cython modules
-  -- by `@webknjaz <https://github.com/sponsors/webknjaz>`__.
+  -- by :user:`webknjaz`.
 
   It will also be reported to Codecov from any non-release CI jobs.
 
@@ -1493,11 +1494,11 @@ Contributor-facing changes
   Editable install produces C-files required for the Cython coverage
   plugin to map the measurements back to the PYX-files.
 
-  `#961 <https://github.com/aio-libs/yarl/issues/961>`__
+  :issue:`961`
 
 - It is now possible to request line tracing in Cython builds using the
-  ``with-cython-tracing`` `PEP 517 <https://peps.python.org/pep-517>`__ config setting
-  -- `@webknjaz <https://github.com/sponsors/webknjaz>`__.
+  ``with-cython-tracing`` :pep:`517` config setting
+  -- :user:`webknjaz`.
 
   This can be used in CI and development environment to measure coverage
   on Cython modules, but is not normally useful to the end-users or
@@ -1520,9 +1521,9 @@ Contributor-facing changes
       $ python -Im pip install -e .
 
   Alternatively, the ``YARL_CYTHON_TRACING=1`` environment variable
-  can be set to do the same as the `PEP 517 <https://peps.python.org/pep-517>`__ config setting.
+  can be set to do the same as the :pep:`517` config setting.
 
-  `#962 <https://github.com/aio-libs/yarl/issues/962>`__
+  :issue:`962`
 
 
 1.9.3 (2023-11-20)
@@ -1531,24 +1532,24 @@ Contributor-facing changes
 Bug fixes
 ---------
 
-- Stopped dropping trailing slashes in ``yarl.URL.joinpath()`` -- by `@gmacon <https://github.com/sponsors/gmacon>`__. (`#862 <https://github.com/aio-libs/yarl/issues/862>`__, `#866 <https://github.com/aio-libs/yarl/issues/866>`__)
-- Started accepting string subclasses in ``yarl.URL.__truediv__()`` operations (``URL / segment``) -- by `@mjpieters <https://github.com/sponsors/mjpieters>`__. (`#871 <https://github.com/aio-libs/yarl/issues/871>`__, `#884 <https://github.com/aio-libs/yarl/issues/884>`__)
-- Fixed the human representation of URLs with square brackets in usernames and passwords -- by `@mjpieters <https://github.com/sponsors/mjpieters>`__. (`#876 <https://github.com/aio-libs/yarl/issues/876>`__, `#882 <https://github.com/aio-libs/yarl/issues/882>`__)
+- Stopped dropping trailing slashes in :py:meth:`~yarl.URL.joinpath` -- by :user:`gmacon`. (:issue:`862`, :issue:`866`)
+- Started accepting string subclasses in :meth:`~yarl.URL.__truediv__` operations (``URL / segment``) -- by :user:`mjpieters`. (:issue:`871`, :issue:`884`)
+- Fixed the human representation of URLs with square brackets in usernames and passwords -- by :user:`mjpieters`. (:issue:`876`, :issue:`882`)
 - Updated type hints to include ``URL.missing_port()``, ``URL.__bytes__()``
-  and the ``encoding`` argument to ``yarl.URL.joinpath()``
-  -- by `@mjpieters <https://github.com/sponsors/mjpieters>`__. (`#891 <https://github.com/aio-libs/yarl/issues/891>`__)
+  and the ``encoding`` argument to :py:meth:`~yarl.URL.joinpath`
+  -- by :user:`mjpieters`. (:issue:`891`)
 
 
 Packaging updates and notes for downstreams
 -------------------------------------------
 
-- Integrated Cython 3 to enable building *yarl* under Python 3.12 -- by `@mjpieters <https://github.com/sponsors/mjpieters>`__. (`#829 <https://github.com/aio-libs/yarl/issues/829>`__, `#881 <https://github.com/aio-libs/yarl/issues/881>`__)
-- Declared modern ``setuptools.build_meta`` as the `PEP 517 <https://peps.python.org/pep-517>`__ build
-  backend in ``pyproject.toml`` explicitly -- by `@webknjaz <https://github.com/sponsors/webknjaz>`__. (`#886 <https://github.com/aio-libs/yarl/issues/886>`__)
-- Converted most of the packaging setup into a declarative ``setup.cfg``
-  config -- by `@webknjaz <https://github.com/sponsors/webknjaz>`__. (`#890 <https://github.com/aio-libs/yarl/issues/890>`__)
-- The packaging is replaced from an old-fashioned ``setup.py`` to an
-  in-tree `PEP 517 <https://peps.python.org/pep-517>`__ build backend -- by `@webknjaz <https://github.com/sponsors/webknjaz>`__.
+- Integrated Cython 3 to enable building *yarl* under Python 3.12 -- by :user:`mjpieters`. (:issue:`829`, :issue:`881`)
+- Declared modern ``setuptools.build_meta`` as the :pep:`517` build
+  backend in :file:`pyproject.toml` explicitly -- by :user:`webknjaz`. (:issue:`886`)
+- Converted most of the packaging setup into a declarative :file:`setup.cfg`
+  config -- by :user:`webknjaz`. (:issue:`890`)
+- The packaging is replaced from an old-fashioned :file:`setup.py` to an
+  in-tree :pep:`517` build backend -- by :user:`webknjaz`.
 
   Whenever the end-users or downstream packagers need to build ``yarl`` from
   source (a Git checkout or an sdist), they may pass a ``config_settings``
@@ -1570,7 +1571,7 @@ Packaging updates and notes for downstreams
       $ python -m build --config-setting=--pure-python=false
 
   Adding ``-w | --wheel`` can force ``pypa/build`` produce a wheel from source
-  directly, as opposed to building an ``sdist`` and then building from it. (`#893 <https://github.com/aio-libs/yarl/issues/893>`__)
+  directly, as opposed to building an ``sdist`` and then building from it. (:issue:`893`)
 
   .. attention::
 
@@ -1578,21 +1579,21 @@ Packaging updates and notes for downstreams
      Later versions dropped the ``--`` prefix, making it just ``pure-python``.
 
 - Declared Python 3.12 supported officially in the distribution package metadata
-  -- by `@edgarrmondragon <https://github.com/sponsors/edgarrmondragon>`__. (`#942 <https://github.com/aio-libs/yarl/issues/942>`__)
+  -- by :user:`edgarrmondragon`. (:issue:`942`)
 
 
 Contributor-facing changes
 --------------------------
 
-- A regression test for no-host URLs was added per `#821 <https://github.com/aio-libs/yarl/issues/821>`__
-  and ``3986`` -- by `@kenballus <https://github.com/sponsors/kenballus>`__. (`#821 <https://github.com/aio-libs/yarl/issues/821>`__, `#822 <https://github.com/aio-libs/yarl/issues/822>`__)
-- Started testing *yarl* against Python 3.12 in CI -- by `@mjpieters <https://github.com/sponsors/mjpieters>`__. (`#881 <https://github.com/aio-libs/yarl/issues/881>`__)
+- A regression test for no-host URLs was added per :issue:`821`
+  and :rfc:`3986` -- by :user:`kenballus`. (:issue:`821`, :issue:`822`)
+- Started testing *yarl* against Python 3.12 in CI -- by :user:`mjpieters`. (:issue:`881`)
 - All Python 3.12 jobs are now marked as required to pass in CI
-  -- by `@edgarrmondragon <https://github.com/sponsors/edgarrmondragon>`__. (`#942 <https://github.com/aio-libs/yarl/issues/942>`__)
-- MyST is now integrated in Sphinx -- by `@webknjaz <https://github.com/sponsors/webknjaz>`__.
+  -- by :user:`edgarrmondragon`. (:issue:`942`)
+- MyST is now integrated in Sphinx -- by :user:`webknjaz`.
 
   This allows the contributors to author new documents in Markdown
-  when they have difficulties with going straight RST. (`#953 <https://github.com/aio-libs/yarl/issues/953>`__)
+  when they have difficulties with going straight RST. (:issue:`953`)
 
 
 1.9.2 (2023-04-25)
@@ -1601,7 +1602,7 @@ Contributor-facing changes
 Bugfixes
 --------
 
-- Fix regression with ``yarl.URL.__truediv__()`` and absolute URLs with empty paths causing the raw path to lack the leading ``/``.
+- Fix regression with :meth:`~yarl.URL.__truediv__` and absolute URLs with empty paths causing the raw path to lack the leading ``/``.
   (`#854 <https://github.com/aio-libs/yarl/issues/854>`_)
 
 
@@ -1623,8 +1624,8 @@ Features
 --------
 
 - Added ``URL.joinpath(*elements)``, to create a new URL appending multiple path elements. (`#704 <https://github.com/aio-libs/yarl/issues/704>`_)
-- Made ``URL.__truediv__()()`` return ``NotImplemented`` if called with an
-  unsupported type — by `@michaeljpeters <https://github.com/sponsors/michaeljpeters>`__.
+- Made :meth:`URL.__truediv__() <yarl.URL.__truediv__>` return ``NotImplemented`` if called with an
+  unsupported type — by :user:`michaeljpeters`.
   (`#832 <https://github.com/aio-libs/yarl/issues/832>`_)
 
 
@@ -1636,7 +1637,7 @@ Bugfixes
   (`#536 <https://github.com/aio-libs/yarl/issues/536>`_)
 - Fixed an issue with update_query() not getting rid of the query when argument is None. (`#792 <https://github.com/aio-libs/yarl/issues/792>`_)
 - Added some input restrictions on with_port() function to prevent invalid boolean inputs or out of valid port inputs; handled incorrect 0 port representation. (`#793 <https://github.com/aio-libs/yarl/issues/793>`_)
-- Made ``yarl.URL.build()`` raise a ``TypeError`` if the ``host`` argument is ``None`` — by `@paulpapacz <https://github.com/sponsors/paulpapacz>`__. (`#808 <https://github.com/aio-libs/yarl/issues/808>`_)
+- Made :py:meth:`~yarl.URL.build` raise a :py:exc:`TypeError` if the ``host`` argument is :py:data:`None` — by :user:`paulpapacz`. (`#808 <https://github.com/aio-libs/yarl/issues/808>`_)
 - Fixed an issue with ``update_query()`` getting rid of the query when the argument
   is empty but not ``None``. (`#845 <https://github.com/aio-libs/yarl/issues/845>`_)
 
@@ -1674,9 +1675,9 @@ Features
 Improved Documentation
 ----------------------
 
-- Fixed broken internal references to ``yarl.URL.human_repr()``.
+- Fixed broken internal references to :meth:`~yarl.URL.human_repr`.
   (`#665 <https://github.com/aio-libs/yarl/issues/665>`_)
-- Fixed broken external references to ``multidict:index`` docs. (`#665 <https://github.com/aio-libs/yarl/issues/665>`_)
+- Fixed broken external references to :doc:`multidict:index` docs. (`#665 <https://github.com/aio-libs/yarl/issues/665>`_)
 
 
 Deprecations and Removals
@@ -2174,13 +2175,13 @@ Features
 0.6.0 (2016-11-07)
 ==================
 
-* Explicitly use UTF8 encoding in ``setup.py`` (#20)
+* Explicitly use UTF8 encoding in :file:`setup.py` (#20)
 * Properly unquote non-UTF8 strings (#19)
 
 0.5.3 (2016-11-02)
 ==================
 
-* Don't use ``typing.NamedTuple`` fields but indexes on URL construction
+* Don't use :py:class:`typing.NamedTuple` fields but indexes on URL construction
 
 0.5.2 (2016-11-02)
 ==================
